sdk/src/main/scala/io/horizen/account/state/ForgerStakeStorageV3.scala

```diff
@@ -0,0 +1,404 @@
+package io.horizen.account.state
+
+import com.google.common.primitives.Bytes
+import io.horizen.account.proposition.{AddressProposition, AddressPropositionSerializer}
+import io.horizen.account.state.ForgerStakeStorageV3.ACCOUNT
+import io.horizen.account.state.NativeSmartContractMsgProcessor.NULL_HEX_STRING_32
+import io.horizen.account.utils.BigIntegerUInt256
+import io.horizen.account.utils.WellKnownAddresses.FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS
+import io.horizen.evm.Address
+import io.horizen.proposition.{PublicKey25519Proposition, VrfPublicKey}
+import io.horizen.utils.BytesUtils
+import sparkz.core.serialization.{BytesSerializable, SparkzSerializer}
+import sparkz.crypto.hash.Blake2b256
+import sparkz.util.serialization.{Reader, Writer}
+
+import java.math.BigInteger
+import scala.collection.mutable.ListBuffer
+
+
+object ForgerStakeStorageV3 {
+
+  val ACCOUNT = FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS
+
+  def getForgerKey(blockSignProposition: PublicKey25519Proposition, vrfPublicKey: VrfPublicKey): Array[Byte] =
+    Blake2b256.hash(Bytes.concat(blockSignProposition.pubKeyBytes(), vrfPublicKey.pubKeyBytes()))
+
+  def getDelegatorKey(delegatorPublicKey: Address): Array[Byte] =
+    Blake2b256.hash(new AddressProposition(delegatorPublicKey).pubKeyBytes())
+
+  def getStakeKey(forgerKey: Array[Byte], delegatorPublicKey: Address): Array[Byte] = {
+    val delegatorKey = getDelegatorKey(delegatorPublicKey)
+    Blake2b256.hash(Bytes.concat(forgerKey, delegatorKey))
+  }
+
+  def addForger(view: BaseAccountStateView,
+                blockSignProposition: PublicKey25519Proposition,
+                vrfPublicKey: VrfPublicKey,
+                rewardShare: Int,
+                rewardAddress: Address,
+                epochNumber: Int,
+                delegatorPublicKey: Address,
+                stakedAmount: BigInteger): Unit = {
```

Comment:
according to the [doc](https://docs.google.com/document/d/1A71qiAENm1SKmclHPczXuTMckM41zClzQiNDRNAFzg0/edit) minimum stake value for creating a forger should be 10 zen - we do not check this condition anywhere in the code, is that ok?

Reply:
Also at 268 - stakedAmount is checked to be >= 0

Reply:
As Paolo said, this class is just the interface to the data in the statedb, so I tried to avoid putting business logic in there as much as possible. I would add this check (or other similar checks) here only if it helped reducing the gas cost of a transaction calling this method, but for now it doesn't seem the case.